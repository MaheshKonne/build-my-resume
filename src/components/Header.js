import React from 'react';
import PropTypes from 'prop-types';
import styled from "styled-components";

const Header = ({heading = 'h1', letterSpacing = 'normal', children, color = '#000'}) => {
    const Component = styled(heading)`
        margin: 0;
        padding: 0;
        color: ${color};
        letter-spacing: ${letterSpacing};
    `;
    return (
        <Component>
            {children}
        </Component>
    );
};

Header.defaultProps = {
    heading: 'h1',
};

Header.propTypes = {
    heading: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ]).isRequired,
};

export default Header;