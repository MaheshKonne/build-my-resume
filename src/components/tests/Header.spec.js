import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Header from "../Header";

describe('<Header />', () => {
    it('should match with snapshot with default values', () => {
        const componentRenderer = renderer.create(<Header>Hello</Header>);
        const component = componentRenderer.toJSON();
        const instance = componentRenderer.root;
        expect(component).toMatchSnapshot();
        // expect(component).toHaveStyleRule('color', '#000');
        // expect(component).toHaveStyleRule('letter-spacing', 'normal');
        expect(instance.findByType('h1')).not.toBeNull();
    });

    it('should match with snapshot with passed props', () => {
        const props = {
            heading: 'h2',
            color: 'red',
            letterSpacing: '10px'
        };
        const componentRenderer = renderer.create(<Header {...props}>Hello</Header>);
        const component = componentRenderer.toJSON();
        const instance = componentRenderer.root;
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule('color', 'red');
        expect(component).toHaveStyleRule('letter-spacing', '10px');
        expect(instance.findByType(props.heading)).not.toBeNull();
    });
});