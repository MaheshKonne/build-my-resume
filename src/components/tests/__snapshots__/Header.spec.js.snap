// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match with snapshot with default values 1`] = `
.c0 {
  margin: 0 0 10px 0;
  padding: 0;
  color: #000;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

<h1
  className="c0"
>
  Hello
</h1>
`;

exports[`<Header /> should match with snapshot with passed props 1`] = `
.c0 {
  margin: 0 0 10px 0;
  padding: 0;
  color: red;
  -webkit-letter-spacing: 10px;
  -moz-letter-spacing: 10px;
  -ms-letter-spacing: 10px;
  letter-spacing: 10px;
}

<h2
  className="c0"
>
  Hello
</h2>
`;
